src/include/OpenImageIO/texture.h

```diff
@@ -187,82 +191,72 @@ class TextureOptions;  // forward declaration
-        MipModeOneLevel,   ///< Use just one mipmap level
-        MipModeTrilinear,  ///< Use two MIPmap levels (trilinear)
-        MipModeAniso,      ///< Use two MIPmap levels w/ anisotropic
-    };
-
-    /// Interp mode determines how we sample within a mipmap level
-    ///
-    enum InterpMode {
-        InterpClosest,      ///< Force closest texel
-        InterpBilinear,     ///< Force bilinear lookup within a mip level
-        InterpBicubic,      ///< Force cubic lookup within a mip level
-        InterpSmartBicubic  ///< Bicubic when magnifying, else bilinear
-    };
+    // Definitions for preserving back compatibility.
+    // These aliases will eventually be deprecated.
+    using Wrap = Tex::Wrap;
+    using MipMode = Tex::MipMode;
+    using InterpMode = Tex::InterpMode;
+    static constexpr Tex::Wrap WrapDefault = Tex::Wrap::Default;
+    static constexpr Tex::Wrap WrapBlack = Tex::Wrap::Black;
+    static constexpr Tex::Wrap WrapClamp = Tex::Wrap::Clamp;
+    static constexpr Tex::Wrap WrapPeriodic = Tex::Wrap::Periodic;
+    static constexpr Tex::Wrap WrapMirror = Tex::Wrap::Mirror;
+    static constexpr Tex::Wrap WrapPeriodicPow2 = Tex::Wrap::PeriodicPow2;
+    static constexpr Tex::Wrap WrapPeriodicSharedBorder = Tex::Wrap::PeriodicSharedBorder;
+    static constexpr Tex::Wrap WrapLast = Tex::Wrap::Last;
+    static constexpr Tex::MipMode MipModeDefault = MipMode::Default;
+    static constexpr Tex::MipMode MipModeNoMIP = MipMode::NoMIP;
+    static constexpr Tex::MipMode MipModeOneLevel = MipMode::OneLevel;
+    static constexpr Tex::MipMode MipModeTrilinear = MipMode::Trilinear;
+    static constexpr Tex::MipMode MipModeAniso = MipMode::Aniso;
+    static constexpr Tex::InterpMode InterpClosest = Tex::InterpMode::Closest;
+    static constexpr Tex::InterpMode InterpBilinear = Tex::InterpMode::Bilinear;
+    static constexpr Tex::InterpMode InterpBicubic = Tex::InterpMode::Bicubic;
+    static constexpr Tex::InterpMode InterpSmartBicubic = Tex::InterpMode::SmartBicubic;
 
 
     /// Create a TextureOpt with all fields initialized to reasonable
     /// defaults.
-    TextureOpt ()
+    OIIO_HOSTDEVICE
+    TextureOpt_v2()
         : firstchannel(0), subimage(0),
-        swrap(WrapDefault), twrap(WrapDefault),
-        mipmode(MipModeDefault), interpmode(InterpSmartBicubic),
-        anisotropic(32), conservative_filter(true),
-        sblur(0.0f), tblur(0.0f), swidth(1.0f), twidth(1.0f),
+        swrap(Wrap::Default), twrap(Wrap::Default), rwrap(Wrap::Default),
+        mipmode(MipMode::Default), interpmode(InterpMode::SmartBicubic),
+        conservative_filter(true), anisotropic(32),
+        sblur(0.0f), tblur(0.0f), rblur(0.0f),
+        swidth(1.0f), twidth(1.0f), rwidth(1.0f),
         fill(0.0f), missingcolor(nullptr),
-        time(0.0f), rnd(-1.0f), samples(1),
-        rwrap(WrapDefault), rblur(0.0f), rwidth(1.0f),
+        rnd(-1.0f),
         colortransformid(0),
         envlayout(0)
```

Comment:
Since you're already changing it, should we replace this with a default constructor and instead do in class initialization?

Reply:
Good idea. Pushed an update.

Reply:
oops, NOW the update is pushed